benches/benchsuite/benches/workspace_initialization.rs

```diff
@@ -0,0 +1,32 @@
+use benchsuite::Bench;
+use cargo::core::Workspace;
+use criterion::{criterion_group, criterion_main, Criterion};
+
+fn workspace_initialization(c: &mut Criterion) {
+    let bench = Bench::setup(env!("CARGO_TARGET_TMPDIR"));
+    let mut group = c.benchmark_group("workspace_initialization");
+    for (ws_name, ws_root) in bench.workspaces() {
+        // This is needed until workspace inheritance is stabilized
+        if ws_name == "rust-ws-inherit" && !cargo_test_support::is_nightly() {
+            continue;
+        } else {
+            let config = bench.make_config(&ws_root);
+            // The resolver info is initialized only once in a lazy fashion. This
+            // allows criterion to skip this workspace if the user passes a filter
+            // on the command-line (like `cargo bench -- workspace_initialization/tikv`).
+            group.bench_function(ws_name, |b| {
+                b.iter(|| Workspace::new(&ws_root.join("Cargo.toml"), &config).unwrap())
```

Comment:
I was wondering if this was the right level as well. I think it is good to track how long it takes to initialize a workspace but it is a very narrow scope. Expanding it to `cargo metadata` or similar might catch changes like #10761. It might be a good idea to add another benchmark for something similar in another PR

Reply:
In general, I lean towards making the primary benchmarks real world (top-level commands) and then using other benchmarks to help gain insight.

Reply:
For now, let's move this forward as-is.  We can iterate over time.

(I meant to resolve this once I had it posted)